samples/subsys/video/capture/src/main.c

```diff
@@ -15,39 +16,85 @@ LOG_MODULE_REGISTER(main);
 
 #define VIDEO_DEV_SW "VIDEO_SW_GENERATOR"
 
+#ifdef CONFIG_VIDEO_MCUX_MIPI_CSI2RX
+#define DISPLAY_FORMAT PIXEL_FORMAT_ARGB_8888
+#else
+#define DISPLAY_FORMAT PIXEL_FORMAT_BGR_565
+#endif
+
+#ifdef CONFIG_MCUX_ELCDIF_PXP
+#define BUFFER_ALIGN 64
+#endif
+
+#if DT_HAS_CHOSEN(zephyr_display)
+void display_setup(const struct device *const display_dev)
+{
+	struct display_capabilities capabilities;
+
+	if (!device_is_ready(display_dev)) {
+		LOG_ERR("Device %s not found", display_dev->name);
+		return;
+	}
+
+	printk("\nDisplay device: %s\n", display_dev->name);
+
+	display_get_capabilities(display_dev, &capabilities);
+
+	printk("- Capabilities:\n");
+	printk("  x_resolution = %u, y_resolution = %u, supported_pixel_formats = %u\n"
+	       "  current_pixel_format = %u, current_orientation = %u\n\n",
+	       capabilities.x_resolution, capabilities.y_resolution,
+	       capabilities.supported_pixel_formats, capabilities.current_pixel_format,
+	       capabilities.current_orientation);
+
+	display_set_pixel_format(display_dev, DISPLAY_FORMAT);
+}
+
+void video_display_frame(const struct device *const display_dev, struct video_buffer *vbuf,
+			 struct video_format fmt)
+{
+	struct display_buffer_descriptor buf_desc;
+
+	buf_desc.buf_size = vbuf->bytesused;
+	buf_desc.width = fmt.width;
+	buf_desc.pitch = buf_desc.width;
+	buf_desc.height = fmt.height;
+
+	display_write(display_dev, 0, 0, &buf_desc, vbuf->buffer);
+
+	display_blanking_off(display_dev);
+}
+#endif
+
 int main(void)
 {
 	struct video_buffer *buffers[2], *vbuf;
 	struct video_format fmt;
 	struct video_caps caps;
-	const struct device *video;
 	unsigned int frame = 0;
 	size_t bsize;
 	int i = 0;
 
-	/* Default to software video pattern generator */
-	video = device_get_binding(VIDEO_DEV_SW);
-	if (video == NULL) {
-		LOG_ERR("Video device %s not found", VIDEO_DEV_SW);
+#if DT_HAS_CHOSEN(zephyr_camera)
+	const struct device *const video_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_camera));
+
+	if (!device_is_ready(video_dev)) {
+		LOG_ERR("%s: video device is not ready", video_dev->name);
 		return 0;
 	}
+#else
```

Comment:
Is the `#else` really required? It looks like the device name is a built-time constant, so resolving the name at runtime with `device_get_binding()` is not required?
I am still unsure about which one to choose in which context.
An alternative would be adding an empty `#else` block with `#error` to hint the user about using a chosen block.

Reply:
`#else` is required because if we don't have a real video device hardware, we use the software generator, but if we have one, we use the hardware, **not** both.